src/python/Utils/TokenManager.py

```diff
@@ -0,0 +1,130 @@
+#-*- coding: utf-8 -*-
+#pylint: disable=W0212
+"""
+File       : TokenManager.py
+Author     : Valentin Kuznetsov <vkuznet AT gmail dot com>
+Description: IAM token manager
+"""
+
+# system modules
+import os
+import ssl
+import time
+import logging
+import traceback
+
+# third part library
+try:
+    import jwt
+except ImportError:
+    traceback.print_exc()
+    jwt = None
+
+# prevent "SSL: CERTIFICATE_VERIFY_FAILED" error
+# this will cause pylint warning W0212, therefore we ignore it above
+ssl._create_default_https_context = ssl._create_unverified_context
+
+
+def getToken(name=None):
+    """
+    Get IAM token either from environment or file name
+    :param name: ether file name containing token or environment name which hold the token value.
+    If not provided it will be assumed to read token from IAM_TOKEN environment.
+    :return: token or None
+    """
+    if name and os.path.exists(name):
+        token = None
+        with open(name) as istream:
+            token = istream.read()
+        return token
+    if name:
+        return os.environ.get(name)
+    return os.environ.get("IAM_TOKEN")
+
+def tokenData(token, url=None, aud=None):
+    """
+    inspect and extract token data
+    :param token: token string
+    :param url: IAM provider URL, by default it is https://cms-auth.web.cern.ch/jwk
+    :param aud: audience string, by default it is https://wlcg.cern.ch/jwt/v1/any
+    """
+    if not token or not jwt:
+        return {}
+    if isinstance(token, str):
+        token = bytes(token, 'utf-8')
+    if not url: # use CMS IAM
+        url = "https://cms-auth.web.cern.ch/jwk"
+    jwksClient = jwt.PyJWKClient(url)
+    signingKey = jwksClient.get_signing_key_from_jwt(token)
+    key = signingKey.key
+    headers = jwt.get_unverified_header(token)
+    alg = headers.get('alg', 'RS256')
+    if not aud: # use WLCG
+        aud = "https://wlcg.cern.ch/jwt/v1/any"
+    data = jwt.decode(
+        token,
+        key,
+        algorithms=[alg],
+        audience=aud,
+        options={"verify_exp": True},
+    )
+    return data
+
+def validToken(token):
+    """
+    validate token expiration timestamp
+    :param token: token string
+    """
+    tokenDict = {}
+    try:
+        tokenDict = tokenData(token)
+    except:
+        traceback.print_exc()
+        raise
+    exp = tokenDict.get('exp', 0)
+    if not exp or exp < time.time():
+        return False
+    return True
+
+
+class TokenManager():
+    """
+    TokenManager class handles IAM tokens
+    """
+    def __init__(self, name=None, url=None, audUrl=None, logger=None):
+        """
+        Token manager reads IAM tokens either from file or env.
+        It caches token along with expiration timestamp.
+        By default the env variable to use is IAM_TOKEN.
+        :param name: string representing either file or env where we should read token from
+        :param url: IAM provider URL, by default it is https://cms-auth.web.cern.ch/jwk
```

Comment:
Same comment as the one above, for these default urls.

Reply:
Another reminder for the default urls (similar to the one provided in the function above).

Reply:
Regarding this comment, please actually consider the one made in the `__init__` method signature.